components/automate-ui/src/app/entities/notification_rules/notification_rule.model.ts

```diff
@@ -67,18 +84,23 @@ export class NotificationRule implements RuleInterface {
     return Object.keys(this.AlertTypeLabels);
   }
 
-  public toRequest(): Object {
-    let target: any = {'url': this.targetUrl};
+  public getTargetTypeKeys(): string[] {
+    return Object.keys(this.TargetTypeLabels);
+  }
+
+  public toRequest(): NotificationRuleTarget {
+    let target: NotificationTarget = { url: this.targetUrl };
     if (this.targetSecretId !== undefined && this.targetSecretId !== '' ) {
       target = {
-        'url': this.targetUrl,
-        'secret_id': this.targetSecretId,
-        'critical_controls_only': this.criticalControlsOnly
+        url: this.targetUrl,
+        secret_id: this.targetSecretId,
+        critical_controls_only: this.criticalControlsOnly
       };
     }
 
     return {
-      'rule': {
+      id: this.id,
+      rule: {
         'name': this.name,
         'event': this.ruleType,
         [this.targetType]: target
```

Comment:
Also, you did not address my question: why do you need to have a variable property name? That seems rather unusual.

If it is really necessary--and that is why you just made `rule` be an object--one way to give `rule` a more defined type is by noting that `this.targetType` only has 3 possible values. So you could create an interface to define `rule` enumerating those 3 values as optional properties.

Reply:
@msorens 
can we add like?
```
export interface NotificationRuleTarget {
  id?: string;
  rule: RuleTarget;
}

export interface RuleTarget { 
  name?: string;
  event?: string;
  targetType?: string;
}

```

Reply:
That is what I think it should be, BUT then this line is not correct:
```
[this.targetType]: target
```
Again, why is this a variable property name?
Shouldn't that line be this?
```
targetType: target
```

Reply:
@msorens 
I tried before adding these changes to this line.
```
export interface NotificationTarget {
  id?: string;
  url?: string;
  secret_id?: string;
  critical_controls_only?: boolean;
}

export interface RuleTarget {
  name?: string;
  event?: string;
  targetType?: NotificationTarget;
}

export interface NotificationRuleTarget {
  id?: string;
  rule: RuleTarget;
}

```
and
```
public toRequest(): NotificationRuleTarget {
    let target: NotificationTarget = { url: this.targetUrl };
    if (this.targetSecretId !== undefined && this.targetSecretId !== '' ) {
      target = {
        url: this.targetUrl,
        secret_id: this.targetSecretId,
        critical_controls_only: this.criticalControlsOnly
      };
    }

    return {
      id: this.id,
      rule: {
        'name': this.name,
        'event': this.ruleType,
        'targetType': target
      }
    };
  }
```
but not work because `[this.targetType]: target` is dynamically changed for example 
if rule type
1. `slack`
![slr](https://user-images.githubusercontent.com/12297653/86325590-e8af3b80-bc5d-11ea-83ce-6cb8ecf537b1.png)
2. `webhook`
![wr](https://user-images.githubusercontent.com/12297653/86325637-febcfc00-bc5d-11ea-911a-cff64d4d783b.png)
3. `service now`
![snr](https://user-images.githubusercontent.com/12297653/86325647-041a4680-bc5e-11ea-88aa-2b031c3a3b22.png)

Reply:
I have pushed [a commit](https://github.com/chef/automate/pull/3927/commits/bd56da61ff0764f12fe6c5d9f4f196b8938ef408) that should clarify the approach I have been suggesting all along. See if that makes sense.

Note that this does introduce an incompatibility with rule.ts, but you said that was going to be removed next. So I leave it to you how to reconcile those, either in this PR or in your next PR.

Reply:
I have updated rule.ts for now because create notification is in progress when I create PR for this at that time I will remove this file.